lib/orbit/import/rfid.ex

```diff
@@ -41,53 +41,61 @@ defmodule Orbit.Import.Rfid do
-          acc ->
-            case Repo.get_by(Employee, badge_number: badge_number) do
-              nil ->
-                acc
+      Repo.transaction(
+        fn ->
+          for {badge_number, badge_serials} <- serials_by_badge, reduce: 0 do
+            acc ->
+              case Repo.get_by(Employee, badge_number: badge_number) do
+                nil ->
+                  acc
 
-              employee ->
-                employee_id = employee.id
-                # Delete any of this employee's badges that were not in the latest import.
-                Repo.delete_all(
-                  from(b in BadgeSerial,
-                    where: b.employee_id == ^employee_id,
-                    where: b.badge_serial not in ^badge_serials
+                employee ->
+                  employee_id = employee.id
+                  # Delete any of this employee's badges that were not in the latest import.
+                  Repo.delete_all(
+                    from(b in BadgeSerial,
+                      where: b.employee_id == ^employee_id,
+                      where: b.badge_serial not in ^badge_serials
+                    )
                   )
-                )
 
-                # And add any new badges for that employee
-                for b <- badge_serials do
-                  %BadgeSerial{employee_id: employee_id, badge_serial: b}
-                  |> BadgeSerial.changeset()
-                  |> Repo.insert(
-                    on_conflict: {:replace_all_except, [:id, :inserted_at]},
-                    conflict_target: :badge_serial
-                  )
-                end
+                  # And add any new badges for that employee
+                  for b <- badge_serials do
+                    %BadgeSerial{employee_id: employee_id, badge_serial: b}
+                    |> BadgeSerial.changeset()
+                    |> Repo.insert(
+                      on_conflict: {:replace_all_except, [:id, :inserted_at]},
+                      conflict_target: :badge_serial
+                    )
+                  end
 
-                acc + length(badge_serials)
-            end
-        end
-      end)
+                  acc + length(badge_serials)
+              end
+          end
+        end,
+        timeout: 120_000
+      )
 
     Logger.info("rfid_import count=#{imported_count}")
   end
 end
 
 defmodule Orbit.Import.RfidWorker do
+  require Logger
   use Oban.Worker, queue: :rfid_import, max_attempts: 3
   alias Orbit.Import.Rfid
 
   @impl Oban.Worker
   @spec perform(Oban.Job.t()) :: :ok
   def perform(%Oban.Job{}) do
-    Rfid.download()
-    |> String.splitter("\n", trim: true)
-    |> CSV.decode!(headers: true, field_transform: &String.trim/1)
-    |> Rfid.import_rows()
+    try do
+      Rfid.download()
+      |> String.splitter("\n", trim: true)
+      |> CSV.decode!(headers: true, field_transform: &String.trim/1)
+      |> Rfid.import_rows()
+    rescue
+      error -> Logger.error(inspect(error))
```

Comment:
Why do you need to squash errors here?

Reply:
We don't have Oban telemetry set up yet (https://app.asana.com/0/616151179860796/1207721363846631/f) so I wasn't seeing anything in Splunk. Needed to do this to quickly see I was exceeding the 15s transaction timeout